src/tube/watcher.js

```diff
@@ -69,10 +69,16 @@ export default class Watcher {
       if (err.message === 'DEADLINE_SOON') {
         await Promise.delay(500);
       }
+      if (err.message.includes('ECONNREFUSED')) {
+        if (onConnectionError) {
+          onConnectionError(err);
+        }
+        await Promise.delay(500);
+      }
```

Comment:
💡interesting part:

We check if the error message contains the word `ECONNREFUSED`.
If it does, we call the new `onConnectionError` callback and wait `500ms`.